packages/datepicker/src/utils/offset.ts

```diff
@@ -26,3 +27,43 @@ export const getNextOffsetDate = (
   }
   return nextDate;
 };
+
+export const getEdgedOffsetDate = (
+  offsetDate: Date,
+  { days, months, years }: DPOffsetValue,
+  dateEdge?: Date,
+): Date => {
+  if (dateEdge) {
+    if (isSame(offsetDate, dateEdge)) return offsetDate;
+    if (days && days !== 0) {
+      return calculateNewDateWithOffset(offsetDate, dateEdge, days, 'date');
+    }
+    if (months && months !== 0) {
+      return calculateNewDateWithOffset(offsetDate, dateEdge, months, 'month');
+    }
+    if (years && years !== 0) {
+      return calculateNewDateWithOffset(offsetDate, dateEdge, years, 'year');
+    }
```

Comment:
Here we can simplify by adding default params during destructuring at the params
```
{ days = 0, months = 0, years = 0 }: DPOffsetValue,
```
Then your check will shrink to `if (days > 0)`, `if (month > 0)` ...

Reply:
done

Reply:
@susosamuel Thanks so much for the fix. @Feshchenko, thanks for the release. It was perfecting timing, thanks to both of you, for my project! There is one little bug that needs to be addressed from this fix. We should also check on the edge if it is either first or last in the array because now the first and last months in the calendar are enabled then after the click event are disabled. My project has asked to fix this in low priority. I can do an external fix in my UI but it would best to handle it in the library. I can make a PR in the coming weeks to fix this small bug.